exercises/bowling/example.rb

```diff
@@ -0,0 +1,105 @@
+class Game
+  class InvalidRoll < StandardError
+    def initialize(msg = "Roll must be between 0 and 10")
+      super
+    end
+  end
+
+  class FramePinsExceeded < StandardError
+    def initialize(msg = "Frame has at most 10 pins.")
+      super
+    end
+  end
+
+  class GameInProgress < StandardError; end
+  class GameHasEnded < StandardError; end
+
+  VERSION = 1
+  TOTAL_FRAMES = 10
+  MINIMUM_ROLL_SCORE = 0
+  MAXIMUM_ROLL_SCORE = 10
+
+  def initialize
+    @frame_scores = []
+    @frame_buffer = []
+  end
+
+  def roll(pins)
+    validate_roll(pins)
+    @frame_buffer << pins
+    adjust_frames
+  end
+
+  def score
+    raise GameInProgress unless @frame_scores.size == TOTAL_FRAMES
+    @frame_scores.reduce(:+)
+  end
+
+  private
+
+  def validate_roll(pins)
+    raise InvalidRoll if pins < MINIMUM_ROLL_SCORE
+    raise InvalidRoll if pins > MAXIMUM_ROLL_SCORE
+
+    if second_roll_in_frame?
+      raise FramePinsExceeded if @frame_buffer[0] + pins > MAXIMUM_ROLL_SCORE
+    end
+
+    raise GameHasEnded if @frame_scores.size == TOTAL_FRAMES
+  end
+
+  def second_roll_in_frame?
+    @frame_buffer.size == 1 && !strike_frame?
+  end
+
+  def strike_frame?
+    @frame_buffer[0] == MAXIMUM_ROLL_SCORE
+  end
+
+  def spare_frame?
+    @frame_buffer[0] + @frame_buffer[1] == MAXIMUM_ROLL_SCORE
+  end
+
+  def open_frame?
+    !strike_frame? && !spare_frame?
+  end
```

Comment:
I have somewhat of a difficult time seeing the single sigil, it is so skinny. but yeah.

I think reading it in your form says "not strike frame and not spare frame", and in my form it says "not strike frame nor spare frame"

I am not sure how most people would say this in an instructional setting.